src/manifolds/Symplectic.jl

```diff
@@ -0,0 +1,889 @@
+SymplecticMatrix(λ::T) where {T<:Number} = SymplecticMatrix{T}(λ)
+
+function SymplecticMatrix(arrays::Vararg{AbstractArray})
+    TS = Base.promote_type(map(eltype, arrays)...)
+    return SymplecticMatrix(one(TS))
+end
+
+@doc raw"""
+    change_representer(::Symplectic, ::EuclideanMetric, p, X)
+    change_representer!(::Symplectic, Y, ::EuclideanMetric, p, X)
+
+Compute the representation of a tangent vector ``χ ∈ T_p\operatorname{Sp}(2n, ℝ)`` s.t.
+````math
+    g_p(c_p(χ), η) = ⟨χ, η⟩^{\text{Euc}} \;∀\; η ∈ T_p\operatorname{Sp}(2n, ℝ).
+````
+with the conversion function
+````math
+    c_p : T_p\operatorname{Sp}(2n, ℝ) \rightarrow T_p\operatorname{Sp}(2n, ℝ), \quad
+    c_p(η) = \frac{1}{2} pp^T η + \frac{1}{2} pQ η^T pQ.
+````
+
+Each of the terms ``c_p^1(η) = p p^T η`` and ``c_p^2(η) = pQ η^T pQ`` from the
+above definition of ``c_p(η)`` are themselves metric compatible in the sense that
+````math
+    c_p^i : T_p\operatorname{Sp}(2n, ℝ) \rightarrow \mathbb{R}^{2n \times 2n}\quad
+    g_p^i(c_p(χ), η) = ⟨χ, η⟩^{\text{Euc}} \;∀\; η ∈ T_p\operatorname{Sp}(2n, ℝ),
+````
+for ``i \in {1, 2}``. However the range of each function alone is not confined to
+``T_p\operatorname{Sp}(2n, ℝ)``, but the convex combination
+````math
+    c_p(η) = \frac{1}{2}c_p^1(η) + \frac{1}{2}c_p^2(η)
+````
+does have the correct range ``T_p\operatorname{Sp}(2n, ℝ)``.
+"""
+change_representer(::Symplectic, ::EuclideanMetric, p, X)
+
+function change_representer!(::Symplectic, Y, ::EuclideanMetric, p, X)
+    Q = SymplecticMatrix(p, X)
+    Y .= (1 / 2) .* p * (p' * X .+ Q * X' * p * Q)
+    return Y
+end
+
+@doc raw"""
+    change_representer(MetMan::MetricManifold{𝔽, Euclidean{Tuple{m, n}, 𝔽}, ExtendedSymplecticMetric},
+                       EucMet::EuclideanMetric, p, X)
+    change_representer!(MetMan::MetricManifold{𝔽, Euclidean{Tuple{m, n}, 𝔽}, ExtendedSymplecticMetric},
+                        Y, EucMet::EuclideanMetric, p, X)
+
+Change the representation of a matrix ``χ ∈ \mathbb{R}^{2n \times 2n}``
+into the inner product space ``(ℝ^{2n \times 2n}, g_p)`` where the inner product
+is given by
+``g_p(χ, η) = \langle p^{-1}χ, p^{-1}η \rangle = \operatorname{tr}(χ^T(pp^T)^{-1}η)``,
+as the extension of the [`RealSymplecticMetric`](@ref) onto the entire embedding space.
+
+By changing the representation we mean to apply a mapping
+````math
+    c_p : \mathbb{R}^{2n \times 2n} \rightarrow \mathbb{R}^{2n \times 2n},
+````
+defined by requiring that it satisfy the metric compatibility condition
+````math
+    g_p(c_p(χ), η) = ⟨p^{-1}c_p(χ), p^{-1}η⟩ = ⟨χ, η⟩^{\text{Euc}}
+        \;∀\; η ∈ T_p\operatorname{Sp}(2n, ℝ).
+````
+In this case, we compute the mapping
+````math
+    c_p(χ) = pp^T χ.
+````
+"""
+function change_representer(
+    MetMan::MetricManifold{𝔽,Euclidean{Tuple{m,n},𝔽},ExtendedSymplecticMetric},
+    EucMet::EuclideanMetric,
+    p,
+    X,
+) where {𝔽,m,n}
+    return change_representer!(MetMan, similar(X), EucMet, p, X)
+end
```

Comment:
Doesn't the generic fallback: https://github.com/JuliaManifolds/Manifolds.jl/blob/f3e1ff5fe567d1df9dac327de22ae913dc8a20f1/src/manifolds/MetricManifold.jl#L166 handle this? We shouldn't have to have this method written.

Reply:
On the other hand, it would make some sense to have `p * p' * X` as the body of the method here, in case we want to work on immutable representations.

Reply:
I've follow the advice of your second comment Mateusz and gone for `return p * p' * X` as the body of the method.